File: Spring/ParkingStatus/src/js/parkingstatus.web/src/services/StatusEventService.ts
import axios from 'axios';

const STATUSEVENT_REST_API_URL : string = 'http://localhost:8080/api/v1/statusevents';

class StatusEventService{

    async getAllStatusEvents(){
        return axios.get(STATUSEVENT_REST_API_URL);
    }

    async getAllStatusEventsOfStatus(statusId : number){
        return axios.get(STATUSEVENT_REST_API_URL+"/status/"+ statusId);
    }

    async getStatusEventById(id: number){
        return axios.get(STATUSEVENT_REST_API_URL+"/"+ id);
    }

    async createStatusEvent(statusEvent: any){
        return axios.post(STATUSEVENT_REST_API_URL, statusEvent);
    }

    updateStatusEvent(id: number, statusEvent: any){}

    deleteStatusEvent(id: number){}

    
}

export default new StatusEventService();

File: Spring/ParkingStatus/src/main/java/com/ParkingStatus/ParkingStatus/Service/StatusEventService.java
package com.ParkingStatus.ParkingStatus.Service;

import com.ParkingStatus.ParkingStatus.DataAccessService.StatusEvent.StatusEventDataAccessService;
import com.ParkingStatus.ParkingStatus.Models.StatusEvent.StatusEvent;
import com.ParkingStatus.ParkingStatus.Models.StatusEvent.StatusEventDate;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class StatusEventService {

    @Autowired
    private final StatusEventDataAccessService statusEventDataAccessService;

    @Autowired
    public StatusEventService(StatusEventDataAccessService statusEventDataAccessService) {
        this.statusEventDataAccessService = statusEventDataAccessService;
    }

    public List<StatusEvent> getAllStatusEvents(){
        return statusEventDataAccessService.selectAllStatusEvents() ;
    }

    public List<StatusEvent> getAllStatusEventsOfStatus(int statusId){
        return statusEventDataAccessService.selectAllStatusEventsOfStatus(statusId);
    }

    public StatusEvent getStatusEventById(int id){
        return statusEventDataAccessService.selectStatusEventById(id);
    }

    public int createStatusEvent(StatusEvent statusEvent){
        return statusEventDataAccessService.insertStatusEvent(statusEvent);
    }

    public int updateStatusEvent(int id, StatusEvent statusEvent){
        return statusEventDataAccessService.updateStatusEvent(id, statusEvent);
    }

    public int deleteStatusEvent(int id){
       return statusEventDataAccessService.removeStatusEvent(id);
    }
}
